docs/dev/README.md

```diff
@@ -0,0 +1,218 @@
+
+To run specific tests, specify a path. For example:
+```sh
+(.venv) $ python3 -m unittest --failfast --verbose test.test_http_client.TestClient.test_connect_http
+```
+
+More path examples:
+*   test - everything under `test/` folder
+*   test.test_http_client - every test in `test_http_client.py`
+*   test.test_http_client.TestClient - every test in `TestClient` class
+*   test.test_http_client.TestClient.test_connect_http - A single test
+
+When creating new tests, note that the names of test files and test functions must be prefixed with `test_`.
+
+### Environment Variables for Tests
+Many tests require an AWS account. These tests are skipped unless
+specific environment variables are set:
+
+*   MQTT
+    *   AWS_TEST_IOT_MQTT_ENDPOINT - AWS account-specific endpoint to connect to IoT core by
+    *   AWS_TEST_TLS_CERT_PATH - file path to certificate used to initialize the TLS context of the MQTT connection
+    *   AWS_TEST_TLS_KEY_PATH - file path to the private key used to initialize the TLS context of the MQTT connection
+    *   AWS_TEST_TLS_ROOT_CERT_PATH - file path to the root CA used to initialize the TLS context of the MQTT connection
+*   PKCS11
+    *   AWS_TEST_PKCS11_LIB - path to PKCS#11 library
+    *   AWS_TEST_PKCS11_PIN - user PIN for logging into PKCS#11 token
+    *   AWS_TEST_PKCS11_TOKEN_LABEL - label of PKCS#11 token
+    *   AWS_TEST_PKCS11_KEY_LABEL - label of private key on PKCS#11 token,
+        which must correspond to the cert at AWS_TEST_TLS_CERT_PATH.
+*   PROXY
+    *   AWS_TEST_HTTP_PROXY_HOST - host address of the proxy to use for tests that make open connections to the proxy
+    *   AWS_TEST_HTTP_PROXY_PORT - port to use for tests that make open connections to the proxy
+    *   AWS_TEST_HTTPS_PROXY_HOST - host address of the proxy to use for tests that make TLS-protected connections to the proxy
+    *   AWS_TEST_HTTPS_PROXY_PORT - port to use for tests that make TLS-protected connections to the proxy
+    *   AWS_TEST_HTTP_PROXY_BASIC_HOST - host address of the proxy to use for tests that make open connections to the proxy with basic authentication
+    *   AWS_TEST_HTTP_PROXY_BASIC_PORT - port to use for tests that make open connections to the proxy with basic authentication
+    *   AWS_TEST_BASIC_AUTH_USERNAME - username to use when using basic authentication to the proxy
+    *   AWS_TEST_BASIC_AUTH_PASSWORD - password to use when using basic authentication to the proxy
+*   S3
+    *   AWS_TEST_S3 - set to any value to enable S3 tests.
+        Unfortunately, at this time these tests can only be run by members of the
+        Common Runtime team, due to use of hardcoded paths. TODO:
+        alter tests so anyone can run them.
+
+## Code Formatting
+
+We use automatic code formatters in this project and pull requests will fail unless
+the code is formatted correctly.
+
+`autopep8` is used for python code. You should have installed via `requirements-dev.txt` above.
```

Comment:
Super minor nitpick: Add `this` after `installed` so it's `You should have installed this via ...`